lib/index.ts

```diff
@@ -67,14 +68,22 @@ class PWMetrics {
 
     for (let runIndex of runs) {
       try {
-        metricsResults[runIndex] = await this.run();
+        const currentMetricResult: MetricsResults = await this.run();
+        currentMetricResult.error = this.resultHasExpectationErrors(currentMetricResult);
```

Comment:
I see you created 2 methods, resultHasExpectationErrors and isAMetricErrorTiming. The first one is really just a 1 line loop through the MetricsResults, I don't think it makes sense to have a function that simply looks an array to call a function. I would suggest you merge both new functions into one.